Codecademy/discourse-codebytes-plugin
# [REACH-650] Save logic works with multiple codebytes in the same post

## Overview
Update the save logic and messaging so that when you have multiple codebytes in one past that you are editing, you can click the save button and have it only update that particular codebyte.

When rendering the editors and their save buttons, we keep track of an index and add that to the message that will get sent to the iframe when save is requested. The response from the iframe to the parent window will now include that index, in addition to the code and language. When the editor is updating the text, it will repeatedly search for codebyte opening tags until it finds the one being referred to by the index.


### PR Checklist
- [x] Related to JIRA ticket: REACH-650

### Notes
Based on top of https://github.com/codecademy-engineering/discourse-codebytes-plugin/pull/15. If it's hard to tell which changes are mine, look at just the commit 13b2843.

Requires https://github.com/codecademy-engineering/static-sites/pull/688

### Testing Instructions
1. Check out the corresponding changes in static sites and get that and Discourse running
2. Draft a post with multiple codebytes. Try some with text in between. Try some with a language specified and some without.
3. Make changes in the editor on the right side. Click save. See that only that codebyte is updated on the left!
4. Do this for each codebyte
![Screen Shot 2021-03-24 at 7 19 58 PM](https://user-images.githubusercontent.com/4821431/112397295-2fefc780-8cd8-11eb-9934-ec55d6eb21d9.png)
![Screen Shot 2021-03-24 at 7 21 01 PM](https://user-images.githubusercontent.com/4821431/112397301-32eab800-8cd8-11eb-99c9-1ac7c67e70c4.png)
